serve/destination.go

```diff
@@ -0,0 +1,146 @@
+package serve
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/internal/pb"
+	"github.com/cloudquery/plugin-sdk/internal/servers"
+	"github.com/cloudquery/plugin-sdk/plugins"
+	"github.com/getsentry/sentry-go"
+	grpczerolog "github.com/grpc-ecosystem/go-grpc-middleware/providers/zerolog/v2"
+	middleware "github.com/grpc-ecosystem/go-grpc-middleware/v2"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+type destinationServe struct {
+	plugin    plugins.DestinationPlugin
+	sentryDSN string
+}
+
+type DestinationOption func(*destinationServe)
+
+func WithDestinationSentryDSN(dsn string) DestinationOption {
+	return func(s *destinationServe) {
+		s.sentryDSN = dsn
+	}
+}
+
+func Destination(plugin plugins.DestinationPlugin, opts ...DestinationOption) {
```

Comment:
Are we happy with the inconsistency that destination plugins use an interface while source plugins use a struct pointer? Should we perhaps make both the same? Asking because now would be the best time to change it 😅

Reply:
I think yes but destinations interface we still have time to change (I remember you had that code somewhere).

Reply:
That's fine, as long as the change we plan on making is to make the destination into a struct pointer as well (and yeah, I had a branch doing that a while back, can dig it up again if we need it)